frontend/test/pytest/test_mlir_plugin_interface.py

```diff
@@ -25,7 +27,60 @@ def test_path_does_not_exists():
     """Test what happens when a pass_plugin is given an path that does not exist"""
 
     with pytest.raises(FileNotFoundError, match="does not exist"):
-        catalyst.apply_pass_plugin("this-path-does-not-exist", "this-pass-also-doesnt-exists")
+        catalyst.passes.apply_pass_plugin(
+            "this-path-does-not-exist", "this-pass-also-doesnt-exists"
+        )
 
     with pytest.raises(FileNotFoundError, match="does not exist"):
-        catalyst.apply_pass_plugin(Path("this-path-does-not-exist"), "this-pass-also-doesnt-exists")
+        catalyst.passes.apply_pass_plugin(
+            Path("this-path-does-not-exist"), "this-pass-also-doesnt-exists"
+        )
+
+
+def test_pass_can_aot_compile():
+    """Can we AOT compile when using apply_pass?"""
+
+    @qml.qjit(target="mlir")
+    @catalyst.passes.apply_pass("some-pass")
+    @qml.qnode(qml.device("null.qubit", wires=1))
+    def example():
+        return qml.state()
+
+    assert example.mlir
+
+
+@pytest.mark.skip()
+def test_pass_plugin_can_aot_compile():
+    """Can we AOT compile when using apply_pass_plugin?
+
+    We can't properly test this because tmp needs to be a valid MLIR plugin.
+    And therefore can only be tested when a valid MLIR plugin exists in the path.
+    """
+
+    with NamedTemporaryFile() as tmp:
+
+        @qml.qjit(target="mlir")
+        @catalyst.passes.apply_pass_plugin(Path(tmp.name), "some-pass")
+        @qml.qnode(qml.device("null.qubit", wires=1))
+        def example():
+            return qml.state()
+
+        assert example.mlir
+
+def test_get_options():
+    """
+      ApplyRegisteredPassOp expects options to be a single StringAttr
+      which follows the same format as the one used with mlir-opt.
+
+    https://mlir.llvm.org/docs/Dialects/Transform/#transformapply_registered_pass-transformapplyregisteredpassop
+
+      Options passed to a pass are specified via the syntax {option1=value1 option2=value2 ...},
+      i.e., use space-separated key=value pairs for each option.
+
+    https://mlir.llvm.org/docs/Tutorials/MlirOpt/#running-a-pass-with-options
+
+    However, experimentally we found that single-options also work without values.
+    """
```

Comment:
Looks like the indentation is slightly off here:
```suggestion
    """
    ApplyRegisteredPassOp expects options to be a single StringAttr
    which follows the same format as the one used with mlir-opt.

    https://mlir.llvm.org/docs/Dialects/Transform/#transformapply_registered_pass-transformapplyregisteredpassop

    Options passed to a pass are specified via the syntax {option1=value1 option2=value2 ...},
    i.e., use space-separated key=value pairs for each option.

    https://mlir.llvm.org/docs/Tutorials/MlirOpt/#running-a-pass-with-options

    However, experimentally we found that single-options also work without values.
    """
```

Reply:
I intended this as a way to denote quoting from those links.

Reply:
Oh I see, isn't there markup for quotes? It's just weird that the docstring starts indented like that, we might want at least a regular summary line before the quotes in that case.

Reply:
> Quoted paragraphs ([ref](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#block-quotes)) are created by just indenting them more than the surrounding paragraphs.